File: topics/control-flow/examples/circle_areas.c
/*
* Program: circle_areas.c
* Displays the Circle Areas for circles with radius
* from 1.0 to 5.0 with increments of 0.1.
*/

#include <stdio.h>

#define PI 3.1415

#define START_RADIUS    1.0
#define END_RADIUS      5.0
#define RADIUS_INC      0.1

double circle_area(double radius)
{
    return PI * radius * radius;
}

int main()
{
    double radius; 
    
    printf("Circle Areas\n");
    printf("-----------------\n");
    
    radius = START_RADIUS;
    
    while (radius <= END_RADIUS)
    {
        printf(" Radius: %4.2f = %4.2f\n", radius, circle_area(radius));
        radius = radius + RADIUS_INC;
    }
    
    printf("-----------------\n");
    
    return 0;
}

File: topics/control-flow/examples/CircleAreas.pas
//
// Program: circle_areas.c
// Displays the Circle Areas for circles with radius 
// from 1.0 to 5.0 with increments of 0.1.
//

program CircleAreas;

const PI = 3.1415;
const START_RADIUS = 1.0;
const END_RADIUS   = 5.0;
const RADIUS_INC   = 0.1;

function CircleArea(radius: Double): Double;
begin
    result := PI * radius * radius;
end;

procedure Main();
var
    radius: Double;
begin
    WriteLn('Circle Areas');
    WriteLn('-----------------');
    radius := START_RADIUS;
    
    while radius <= END_RADIUS do
    begin
        WriteLn(' Radius: ', radius:4:2, ' = ', CircleArea(radius):4:2);
        radius := radius + RADIUS_INC;
    end;
    WriteLn('-----------------');
end;

begin
    Main();
end.
